assets/js/aplus.js

```diff
@@ -584,65 +588,182 @@ $.fn.highlightCode = function(options) {
+          filterCell.append(filterInput);
+          filterRow.append(filterCell);
+        }
+        this.element.find('thead').append(filterRow);
+      },
 
-          var self = this;
-          var filterDelay = function(event) {
-            var input = $(this);
-            clearTimeout(self.timeout);
-            self.timeout = setTimeout(function() {
-              self.filterColumn(input);
-            }, 500);
-          };
-
-          this.filters = [];
-          var filterRow = $('<tr></tr>');
-          for (var i = 0; i < columnCount; i++) {
-            this.filters.push('');
-            var filterInput = $('<input type="text" data-column="'+i+'">')
-              .on('keyup', filterDelay).on('change', filterDelay);
-            var filterCell = $('<td></td>');
-            filterCell.append(filterInput);
-            filterRow.append(filterCell);
+      validateRegex: function(pattern) {
+        var parts = pattern.split('/');
+        var regex = pattern;
+        var options = "";
+
+        if (parts.length > 1) {
+            regex = parts[1];
+            options = parts[2];
+        }
+        try {
+            // If no error is thrown here, the pattern is valid
+            new RegExp(regex, options);
+            return true;
+        }
+        catch(e) {
+            return false;
+        }
+      },
+
+      filterGetType: function(query) {
+          if (!query) {
+            // Empty query, no filtering
+            return [undefined, ""];
+          }
+
+          if (query.startsWith("<")) {
+            if (query.startsWith("<=")) {
+                return [this.lteFilter, query.slice(2)];
+            }
+            else {
+                return [this.ltFilter, query.slice(1)];
+            }
+          }
+
+          else if (query.startsWith(">")) {
+            if (query.startsWith(">=")) {
+                return [this.gteFilter,  query.slice(2)];
+            }
+            else {
+                return [this.gtFilter, query.slice(1)];
+            }
+          }
+
+          else if (query.startsWith("==")) {
+              return [this.eFilter, query.slice(2)];
+          }
+
+          else if (query.startsWith("!=")) {
+              return [this.neFilter,  query.slice(2)];
           }
-          this.element.find('thead').append(filterRow);
-        },
 
-        filterColumn: function(input) {
+          else if (query.match(/\//g) && query.match(/\//g).length >= 2) {
+            if (this.validateRegex(query)) {
+              return [this.regexFilter, query];                      
+            }
+            else {
+              return [this.normalFilter, query];
+            }
+          }
+
+          else {
+            return [this.normalFilter, query];
+          }
+      },
+
+      lteFilter: function(text, query) {
+        return isNaN(text) || isNaN(query) ? text <= query : 
```

Comment:
`isNaN(query)` and `parseFloat(query)` should be done in `filterGetType`

Reply:
I see the problem that the query is `<10` and the sell value is `foobar`. Good question is what should be done then. Should we show warning in the filter box "you can't use numeric filters with non-numeric data" or so.

Reply:
Check for isNaN is done in filterGetType. Still unsure if there are need for the extra warning. Obviously not hard to implement.

Currently the query 10 would be turned to number 10, since it has only digits. If cell value is "foobar", then comparison "foobar" < 10 tries to convert "foobar" to number, giving NaN and the evaluation is false always. I.E. both "foobar" > 10 and "foorbar" < 10 results in false currently.